docs/source/fx.rst

```diff
@@ -10,7 +10,167 @@ Overview
 Writing Transformations
 -----------------------
 
-TODO
+What is a FX transform? Essentially, it's a function that looks like this.
+
+::
+
+    def transform(m: nn.Module) -> nn.Module:
+        fx_model: GraphModule = fx.symbolice_trace(m)
+        new_model = ...
+        return new_model
+
+Your transform will take in an :class:`torch.nn.Module`, convert it into a
+:class:`GraphModule` with :meth:``symbolic_trace``, and return a new
+``nn.Module``. You should think of the ``nn.Module`` that your FX transform
+returns as identical to a regular ``nn.Module`` -- you can pass it to another
+FX transform, you can pass it to TorchScript, or you can
+run it. Ensuring that the inputs and outputs of your FX transform are a
+``nn.Module`` will allow for composability.
+
+Given that you’ve passed in an ``nn.Module`` that has been traced into a
+graph, there are now two primary approaches you can take to building a new
+graph.
+
+Graph Manipulation
+------------------
+
+One approach to building this new graph is to simply transform your old
+one. To aid in this, we can simply take the graph we obtain from
+symbolic tracing and modify it. For example, let’s say we desire to
+replace ``torch.add`` with ``torch.mul``.
+
+::
+
+    # Sample module
+    class M(torch.nn.Module):
+        def forward(self, x, y):
+            return torch.add(x, y)
+
+    def transform(m: nn.Module) -> nn.Module:
+        fx_model: GraphModule = fx.symbolic_trace(m)
+        # FX represents its graph as an ordered list of nodes, so we can iterate through them.
+        for node in fx_model.graph.nodes:
+            # Checks if we're calling a function (i.e: torch.add)
+            if node.op == 'call_function':
+                # The target attribute is the function that call_function calls.
+                if node.target == torch.add:
+                    node.target = torch.mul
+
+        traced.lint() # Does some checks to make sure the graph is well-formed.
+        traced.recompile() # regenerates the python code that corresponds to the graph.
+
+We can also do more involved graph rewrites, such as appending a ReLU
```

Comment:
Seems like "We can also do more involved graph rewrites, such as appending a ReLU..." should go after the bit about the API reference since it sets up the next code example